api/src/main/java/io/minio/MinioClient.java

```diff
@@ -4869,6 +4857,235 @@ public SelectResponseStream selectObjectContent(
+   * @throws IllegalArgumentException throws to indicate invalid argument passed.
+   * @throws InsufficientDataException thrown to indicate not enough data available in InputStream.
+   * @throws InternalException thrown to indicate internal library error.
+   * @throws InvalidBucketNameException thrown to indicate invalid bucket name passed.
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws InvalidResponseException thrown to indicate S3 service returned invalid or no error
+   *     response.
+   * @throws IOException thrown to indicate I/O error on S3 operation.
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws XmlParserException thrown to indicate XML parsing error.
+   */
+  public Tags getObjectTags(String bucketName, String objectName)
+      throws ErrorResponseException, IllegalArgumentException, InsufficientDataException,
+          InternalException, InvalidBucketNameException, InvalidKeyException,
+          InvalidResponseException, IOException, NoSuchAlgorithmException, XmlParserException {
+    checkBucketName(bucketName);
+    checkObjectName(objectName);
+
+    Map<String, String> queryParamMap = new HashMap<>();
+    queryParamMap.put("tagging", "");
+
+    Response response = executeGet(bucketName, objectName, null, queryParamMap);
+    try (ResponseBody body = response.body()) {
+      return Xml.unmarshal(Tags.class, body.charStream());
+    }
+  }
+
+  /**
+   * Sets tags to an object.
+   *
+   * <pre>Example:{@code
+   * Map<String, String> map = new HashMap<>();
+   * map.put("Project", "Project One");
+   * map.put("User", "jsmith");
+   * minioClient.setObjectTags("my-bucketname", Tags.newObjectTags(map));
+   * }</pre>
+   *
+   * @param bucketName Name of the bucket.
+   * @param objectName Object name in the bucket.
+   * @param tags {@link Tags} to be set.
+   * @throws ErrorResponseException thrown to indicate S3 service returned an error response.
+   * @throws IllegalArgumentException throws to indicate invalid argument passed.
+   * @throws InsufficientDataException thrown to indicate not enough data available in InputStream.
+   * @throws InternalException thrown to indicate internal library error.
+   * @throws InvalidBucketNameException thrown to indicate invalid bucket name passed.
+   * @throws InvalidKeyException thrown to indicate missing of HMAC SHA-256 library.
+   * @throws InvalidResponseException thrown to indicate S3 service returned invalid or no error
+   *     response.
+   * @throws IOException thrown to indicate I/O error on S3 operation.
+   * @throws NoSuchAlgorithmException thrown to indicate missing of MD5 or SHA-256 digest library.
+   * @throws XmlParserException thrown to indicate XML parsing error.
+   */
+  public void setObjectTags(String bucketName, String objectName, Tags tags)
```

Comment:
IMO this PR can be reviewed once  `setObjectTags` uses builder pattern similar to `setBucketTags`